### PR TITLE
overrides: make the overrides tegra specific 

### PR DESCRIPTION
**Pull request by quaresmajose**

### CODE REVIEW

File: external/openembedded-layer/recipes-extended/haveged/haveged_%.bbappend [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
-SYSTEMD_PACKAGES = "${PN}"
-SYSTEMD_SERVICE:${PN} = "haveged.service"
+SYSTEMD_PACKAGES:tegra = "${PN}"
+SYSTEMD_SERVICE:${PN}:tegra = "haveged.service"
 
 # Based on init.d/service.redhat from 1.9.14
 # https://raw.githubusercontent.com/jirka-h/haveged/v1.9.14/init.d/service.redhat
-SRC_URI += "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'file://haveged.service', '', d)}"
+SRC_URI:tegra += "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'file://haveged.service', '', d)}"
```
Comment (madisongh) [inline external/openembedded-layer/recipes-extended/haveged/haveged_%.bbappend:9]: This should be `SRC_URI:append:tegra =`, and a blank will be needed at the front of the string.

File: external/openembedded-layer/recipes-multimedia/v4l2apps/v4l-utils_%.bbappend [hunk 2]
```diff
@@ -20,6 +20,6 @@ FILES_libv4l:remove:tegra = "${libdir}/libv4l/plugins/*.so"
 
 TEGRA_PLUGINS ?= ""
 TEGRA_PLUGINS:tegra = "tegra-libraries-multimedia-v4l"
-RRECOMMENDS:libv4l += "${TEGRA_PLUGINS}"
+RRECOMMENDS:libv4l:tegra += "${TEGRA_PLUGINS}"
```
Comment (madisongh) [inline external/openembedded-layer/recipes-multimedia/v4l2apps/v4l-utils_%.bbappend:23]: Is this change needed?  For non-tegra platforms, the value of `TEGRA_PLUGINS` will be null anyway.

File: external/qt5-layer/recipes-qt/qt5/qtbase_%.bbappend [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
 
-SRC_URI += "file://0001-eglfs-add-a-default-framebuffer-to-NVIDIA-eglstreams.patch \
+SRC_URI:tegra += "file://0001-eglfs-add-a-default-framebuffer-to-NVIDIA-eglstreams.patch \
```
Comment (madisongh) [inline external/qt5-layer/recipes-qt/qt5/qtbase_%.bbappend:3]: This should use `:append:tegra =` as well

File: external/virtualization-layer/recipes-containers/docker/docker-ce_%.bbappend [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
         install -d ${D}${sysconfdir}/init.d
         install -m 0755 ${WORKDIR}/docker.init ${D}${sysconfdir}/init.d/docker.init
     fi
 }
 
-RRECOMMENDS:${PN}:append = " \
+RRECOMMENDS:${PN}:append:tegra = " \
     kernel-module-br-netfilter \
     kernel-module-esp4 \
     kernel-module-ip-vs \
```
Comment (madisongh) [inline external/virtualization-layer/recipes-containers/docker/docker-ce_%.bbappend:11]: With the above changes, we'll also need
```
PACKAGE_ARCH:tegra = "${TEGRA_PKGARCH}"
```

**Review by madisongh — CHANGES_REQUESTED** [review]
Thanks for catching these, and for the patches.  A few tweaks are needed, but otherwise looks OK.

File: external/openembedded-layer/recipes-extended/haveged/haveged_%.bbappend (hunk 1 above)
Comment (quaresmajose) [reply external/openembedded-layer/recipes-extended/haveged/haveged_%.bbappend:9]: right, I will change it to `':append:tegra = " '`

File: external/qt5-layer/recipes-qt/qt5/qtbase_%.bbappend (hunk 3 above)
Comment (quaresmajose) [reply external/qt5-layer/recipes-qt/qt5/qtbase_%.bbappend:3]: right, I will change it to `':append:tegra = " '`

File: external/virtualization-layer/recipes-containers/docker/docker-ce_%.bbappend (hunk 4 above)
Comment (quaresmajose) [reply external/virtualization-layer/recipes-containers/docker/docker-ce_%.bbappend:11]: I will make it arch specific

Comment (quaresmajose) [reply external/virtualization-layer/recipes-containers/docker/docker-ce_%.bbappend:11]: This append was what caught my attention and then I took the opportunity to correct the others

**Review by madisongh — APPROVED** [review]

**Comment by madisongh** [comment]
Thanks!
